homework_1/02_ProgressiveTax/taxes/taxes.go

```diff
@@ -0,0 +1,60 @@
+package taxes
+
+import "github.com/pkg/errors"
+
+type taxBrackets struct {
+	brackets []float32
+	taxes    []float32
+}
```

Comment:
How about 
```suggestion
type taxBracket struct {
	brackets float32
	taxes    float32
}
```

And then use a slice of this struct in your functions?
This way you don't need to assert that `len(brackets) == len(taxes)`

Reply:
For me, this made more sense conceptually, as a single bracket defined like that makes no sense.

Reply:
Not sure what you mean exactly?

Reply:
If there would exist taxBracket {10000, 0.2} it would't mean anything without knowing about other taxBrackets. It would mean something if each tax bracket would have starting and ending amount, but for me it only makes sense if they are all together.

Reply:
Sure, but that's besides the point here. Every bracket needs to have its threshold and the tax to apply.
So having a custom struct
```
type taxBracket struct {
  threshold float32
  tax float32
}
```

and then using `[]taxBracket` as your "configuration" makes it a lot easier to reason about it and it makes your validation code simpler (i.e. you don't need to assert `len(taxes) == len(brackets)`)

Reply:
I refactored as you suggested. :smile: